packages/stream_chat_persistence/lib/src/stream_chat_persistence_client.dart

```diff
@@ -0,0 +1,222 @@
+import 'package:stream_chat/stream_chat.dart';
+
+import 'db/moor_chat_database.dart';
+import 'db/shared/shared_db.dart';
+
+/// Various connection modes on which [StreamChatPersistenceClient] can work
+enum ConnectionMode {
+  /// Connects the [StreamChatPersistenceClient] on a regular/default isolate
+  regular,
+
+  /// Connects the [StreamChatPersistenceClient] on a background isolate
+  background,
+}
+
+/// A [MoorChatDatabase] based implementation of the [ChatPersistenceClient]
+class StreamChatPersistenceClient extends ChatPersistenceClient {
+  /// Creates a new instance of the stream chat persistence client
+  StreamChatPersistenceClient({
+    /// Connection mode on which the client will work
+    ConnectionMode connectionMode = ConnectionMode.regular,
+    Level logLevel = Level.WARNING,
+  })  : assert(connectionMode != null),
+        assert(logLevel != null),
+        _connectionMode = connectionMode,
+        _logger = Logger.detached('💽')..level = logLevel;
+
+  MoorChatDatabase _db;
+  final Logger _logger;
+  final ConnectionMode _connectionMode;
+
+  @override
+  Future<void> connect(String userId) async {
+    if (_db != null) {
+      throw Exception(
+        'An instance of StreamChatDatabase is already connected.\n'
+        'disconnect the previous instance before connecting again.',
+      );
+    }
+    switch (_connectionMode) {
+      case ConnectionMode.regular:
+        _logger.info('Connecting on a regular isolate');
+        _db = MoorChatDatabase(userId);
+        return;
+      case ConnectionMode.background:
+        _logger.info('Connecting on background isolate');
+        _db = await SharedDB.constructOfflineStorage(userId);
+        return;
+    }
+  }
+
+  @override
+  Future<Event> getConnectionInfo() {
```

Comment:
Consider adding doc comments to these functions.

Reply:
Overridden function and member copies the doc comments from the parent class by default, So I don't think it's necessary here.